File: src/app/lib/src/dtos/input_models/update_post_input_model.dart
import "package:dio/dio.dart";
import "package:image_picker/image_picker.dart";

class UpdatePostInputModel {
  final String? content;
  final String? tablature;
  final List<int> interestsIds;
  final List<XFile> newMedias;
  final List<int> mediasToRemove;

  UpdatePostInputModel({
    this.content,
    this.tablature,
    required this.interestsIds,
    required this.mediasToRemove,
    required this.newMedias
  });

  FormData toFormData() {
    final FormData formData = FormData.fromMap({
      "content": this.content,
      "tablature": this.tablature,
      "interestsIds": this.interestsIds,
      "mediasToRemove": this.mediasToRemove
    });
    formData.files.addAll(this.newMedias.map((media) => MapEntry("newMedias", MultipartFile.fromFileSync(media.path))));
    return formData;
  }
}

File: src/API/Microsservices/Post/Sonorus.Post.Application/Commands/UpdatePost/UpdatePostInputModel.cs
using Microsoft.AspNetCore.Http;

namespace Sonorus.Post.Application.Commands.UpdatePost;

public class UpdatePostInputModel {
    public string? Content { get; set; }
    public string? Tablature { get; set; }
    public ICollection<long> InterestsIds { get; set; } = [];
    public ICollection<long> MediasToRemove { get; set; } = [];
    public ICollection<IFormFile> NewMedias { get; set; } = [];
}